### PR TITLE
fix: avoid endless recursion

### PR DESCRIPTION
**Pull request by nhcarrigan**
Checklist:

- [X] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [X] My pull request has a [descriptive title](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) (**not** a vague title like `Update index.md`)

Closes #XXXXX

In the current state, the recursion get stuck in a loop if an unsolvable puzzle string is submitted. @ojeytonwilliams and I discussed an approach to fixing this with an arbitrary timeout, but after further thought I decided counting the recursive calls might be better.

A quick Google search gave me the "world's hardest sudoku", which the solver took 49000+ recursive calls to solve. By counting and capping the recursive calls at 50000, this PR will add an assumption that additional calls mean the puzzle is unsolvable and back out of the recursion.

Passing an unsolvable string with this change no longer locks up the server, but instead properly displays "puzzle cannot be solved" in the front end.

### CODE REVIEW

File: apps/sudoku-solver/controllers/sudoku-solver.js [hunk 1]
```diff
@@ -194,6 +196,15 @@ class SudokuSolver {
   // all unassigned locations in such a way to meet the requirements
   // for Sudoku solution (non-duplication across rows, columns, and boxes)
   solveSudoku(grid) {
+    this._recursions++;
+    console.log(this._recursions);
+    // Tested the recursion count for the "hardest sudoku" puzzle
+    // Puzzle was fetched from https://www.conceptispuzzles.com/index.aspx?uri=info/article/424
+    // Recursion count was 49559 - rounded to 50000 for nice number + a bit of buffer.
+    // Anything exceeding this should be an unsolvable puzzle.
```
Comment (nhcarrigan) [inline apps/sudoku-solver/controllers/sudoku-solver.js:204]: I hate magic numbers, so documented the reason for 50000 in the comments.

File: apps/sudoku-solver/controllers/sudoku-solver.js [hunk 2]
```diff
@@ -194,6 +196,15 @@ class SudokuSolver {
   // all unassigned locations in such a way to meet the requirements
   // for Sudoku solution (non-duplication across rows, columns, and boxes)
   solveSudoku(grid) {
+    this._recursions++;
+    console.log(this._recursions);
```
Comment (ojeytonwilliams) [inline apps/sudoku-solver/controllers/sudoku-solver.js:200]:
```suggestion
```

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
Errant log aside, LGTM

File: apps/sudoku-solver/controllers/sudoku-solver.js (hunk 2 above)
Comment (nhcarrigan) [reply apps/sudoku-solver/controllers/sudoku-solver.js:200]: I knew I'd forget this.

**Review by ojeytonwilliams — APPROVED** [review]

**Pull request merged**